client/rest/src/main/java/org/elasticsearch/client/Node.java

```diff
@@ -175,42 +176,35 @@ public int hashCode() {
      * Role information about an Elasticsearch process.
      */
     public static final class Roles {
-        private final boolean masterEligible;
-        private final boolean data;
-        private final boolean ingest;
-
-        public Roles(boolean masterEligible, boolean data, boolean ingest) {
-            this.masterEligible = masterEligible;
-            this.data = data;
-            this.ingest = ingest;
+
+        private final Set<String> roles;
+
+        public Roles(final Set<String> roles) {
+            this.roles = new TreeSet<>(roles);
         }
 
         /**
          * Teturns whether or not the node <strong>could</strong> be elected master.
          */
         public boolean isMasterEligible() {
-            return masterEligible;
+            return roles.contains("master");
```

Comment:
Out of curiosity, should we make these constants since they're hardcoded as strings all over the place? (in separate work, not this PR)